semgrep/semgrep/rule.py

```diff
@@ -247,6 +246,7 @@ def has_runnable_rule(d: Dict[str, Any]) -> bool:
 
 
 def rule_without_metadata(rule: Rule) -> Rule:
-    rule2 = deepcopy(rule)
```

Comment:
what are the implications of now doing a shallow copy?

Reply:
Nothing! We only ever use this to create a read-only dictionary key.